### ISSUE TITLE
Drop support for cs:implements metadata

### ISSUE THREAD

**Issue by pepone** [zeroc-ice/ice#3247]
See also #3246

https://doc.zeroc.com/ice/3.7/the-slice-language/slice-metadata-directives#id-.SliceMetadataDirectivesv3.7-cs:implements:type

Given than C# generated types are always partial types, you can manually declare this additional interfaces in a separate partial struct/class that also provides the implementation.

For example:
```
// In Foo.cs (generated by slice2cs)
partial class Foo {}

// In Foo.Disposable.cs (written by the user)
partial class Foo : IDisposable
{
    public void Dispose()
    {
        throw new NotImplementedException();
    }
}
```

**Comment by InsertCreativityHere** [comment]
This proposal is absolutely doable, but I don't see any advantage to it.
If we were designed this from the ground up, then sure, this proposal is fine (which is why icerpc works exactly as described).

But, Ice does support this metadata, and users are probably already using it.
Removing this feature, and making users add the attribute 'by-hand' is just annoying to them.
And it's not like there's any benefit to adding metadata via a partial, they are functionally equivalent.

**Comment by pepone** [comment]
The only example we provide is for a local Slice, which makes me thing that was the most common use case. Then just adding and interface to the list of inherited interfaces doesn't seems that useful, you still need to provide an implementation.

**Comment by bernardnormier** [comment]
I agree with Jose, I doubt `cs:implements` is used much if at all.

**Comment by InsertCreativityHere** [comment]
Indeed, searching on GitHub shows that the only user of this metadata was us (for local Slice).
I'll remove it.

**Comment by bernardnormier** [comment]
I would also drop `java:implements`:
https://doc.zeroc.com/ice/3.7/the-slice-language/slice-metadata-directives#id-.SliceMetadataDirectivesv3.7-java:implements:type
